jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java

```diff
@@ -18,179 +18,394 @@
     {
-        EnumSet<HttpComplianceSection> sections;
-        String[] elements = spec.split("\\s*,\\s*");
-        int i=0;
-        
-        switch(elements[i])
-        {       
-            case "0":
-                sections = EnumSet.noneOf(HttpComplianceSection.class);
-                i++;
-                break;
-                
-            case "*":
-                i++;
-                sections = EnumSet.allOf(HttpComplianceSection.class);
-                break;
-                
-            case "RFC2616":
-                sections = EnumSet.complementOf(EnumSet.of(
-                HttpComplianceSection.NO_FIELD_FOLDING,
-                HttpComplianceSection.NO_HTTP_0_9));
-                i++;
-                break;
-                
-            case "RFC7230":
-                i++;
-                sections = EnumSet.allOf(HttpComplianceSection.class);
-                break;
-
-            default:
-                sections = EnumSet.noneOf(HttpComplianceSection.class);
-                break;
-        }
-
-        while(i<elements.length)
-        {
-            String element = elements[i++];
-            boolean exclude = element.startsWith("-");
-            if (exclude)
-                element = element.substring(1);
-            HttpComplianceSection section = HttpComplianceSection.valueOf(element);
-            if (section==null)
-            {
-                LOG.warn("Unknown section '"+element+"' in HttpCompliance spec: "+spec);
-                continue;
-            }
-            if (exclude)
-                sections.remove(section);
-            else
-                sections.add(section);
-
-        }
-        
-        return sections;
+        return HttpCompliance.rfc7230Builder().build();
     }
-    
-    private final static Map<HttpComplianceSection,HttpCompliance> __required = new HashMap<>();
-    static
+
+    /**
+     * <p>
+     * Builder based on obsolete <a href="https://tools.ietf.org/html/rfc2616">RFC 2616: HTTP/1.1</a>.
+     * </p>
+     *
+     * <table>
+     * <caption>RFC 2616 Defaults</caption>
+     * <thead>
+     * <tr><td>Configurable</td><td>Default Value</td></tr>
+     * </thead>
+     * <tbody>
+     * <tr><td>allowCaseInsensitiveFieldNames</td><td>true</td></tr>
+     * <tr><td>allowCaseInsensitiveRequestMethod</td><td>true</td></tr>
+     * <tr><td>allowHttp09</td><td>true</td></tr>
+     * <tr><td>allowMultiLineFieldValue</td><td>true</td></tr>
+     * <tr><td>allowMultipleContentLengths</td><td>true</td></tr>
+     * <tr><td>allowTransferEncodingWithContentLength</td><td>true</td></tr>
+     * <tr><td>allowWhitespaceAfterFieldName</td><td>true</td></tr>
+     * <tr><td>requireColonAfterFieldName</td><td>true</td></tr>
+     * <tr><td>useCaseInsensitiveFieldValueCache</td><td>true</td></tr>
+     * </tbody>
+     * </table>
+     *
+     * @return new Builder with configuration set to RFC2616 Strict behavior.
+     */
+    public static Builder rfc2616Builder()
```

Comment:
I like these methods - get a builder with a specific starting point that can then be modified!